feat: move dependency crds to harvester-crd folder

Yu-Jack opened the pull request:
<!-- 
!IMPORTANT!
Please do not create a Pull Request without creating an issue first.
-->

#### Problem:
The problem is pretty clear in original issue. So, I skip it here.

#### Solution:

We should consider two situations: installation and upgrading.

There are different solutions here:

- Use bindata to manage CRDs
- Create another managedChart to manage CRDs
- Apply CRDs independently [here](https://github.com/harvester/harvester/blob/71f3505c9bf9d956267065846642098850ebea23/package/upgrade/upgrade_manifests.sh#L1358)

Each Drawback:

- Order issue. It's hard to guarantee that other components can use latest CRDs. With this way, CRDs are updated after Harvester starts up. It might have potential bug while upgrading.
- Too many managedCharts. We've had a lot of workaround ways to deal with managedChart [here](https://github.com/harvester/harvester/blob/master/package/upgrade/upgrade_manifests.sh#L1358). I don't think we might want more.
- Those CRDs are still not in the control. They don't belong to any management flow.

The quick way is to put those dependency CRDs under harvester-crd. Then, with [takeOwnerShip field](https://github.com/harvester/harvester-installer/blob/90668931a78c860f309487fe100be5ef868c3f81/pkg/config/templates/rancherd-10-harvester.yaml#L235), it helps us add labels and annotations back and manages those CRDs.

***However, we won't put cdi under harvester-crd because cdi-operator will manage CRD status. For example, they will [delete old version here](https://github.com/kubevirt/containerized-data-importer/blob/e136558c999acce03291849aacc8154026e58445/pkg/operator/controller/callbacks.go#L61-L62). This is also mentioned in https://github.com/harvester/harvester/issues/7912#issuecomment-2743944344.***

#### Related Issue(s):
#8163 

#### Test plan:

1. Install new master-head or v1.5-head
2. Upgrade from v1.4.3 to v1.5-head
3. Upgrade from v1.5 to master-head or v1.5-head

You could add a new dummy field in those CRDs, then test it.

And before upgrade, please ensure that `cluster-repo` has been patched.

```
sudo mount -o loop ~/Downloads/harvester-master-cpu-pinning/harvester-master-amd64.iso /mnt/iso
scp -O /mnt/iso/bundle/harvester/images/harvester-repo-images-master.tar.zst rancher@xxxx:/home/rancher

# In node
docker load < /home/rancher/harvester-repo-images-master.tar.zst
# then, kill the cluster-repo pod
```

With this, your master-build iso can work.

Regardless of cases, those CRDs should contain specific labels and annotation like below:

- helm.sh/resource-policy: keep
- meta.helm.sh/release-name: harvester-crd
- meta.helm.sh/release-namespace: harvester-system
- objectset.rio.cattle.io/id: default-mcc-harvester-crd-cattle-fleet-local-system
- app.kubernetes.io/managed-by: Helm
- objectset.rio.cattle.io/hash: d4a83267ddde6a8769c04362d4a0e5605db9baa7

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    helm.sh/resource-policy: keep
    meta.helm.sh/release-name: harvester-crd
    meta.helm.sh/release-namespace: harvester-system
    objectset.rio.cattle.io/id: default-mcc-harvester-crd-cattle-fleet-local-system
  creationTimestamp: "2025-05-06T14:05:28Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
    objectset.rio.cattle.io/hash: d4a83267ddde6a8769c04362d4a0e5605db9baa7
    operator.kubevirt.io: ""
  name: kubevirts.kubevirt.io
  resourceVersion: "623701"
  uid: ea688774-d544-4f5d-bc61-a4abf76175d9
---
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.14.0
    helm.sh/resource-policy: keep
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apiextensions.k8s.io/v1","kind":"CustomResourceDefinition","metadata":{"annotations":{"controller-gen.kubebuilder.io/version":"v0.14.0"},"name":"overlappingrangeipreservations.whereabouts.cni.cncf.io"},"spec":{"group":"whereabouts.cni.cncf.io","names":{"kind":"OverlappingRangeIPReservation","listKind":"OverlappingRangeIPReservationList","plural":"overlappingrangeipreservations","singular":"overlappingrangeipreservation"},"scope":"Namespaced","versions":[{"name":"v1alpha1","schema":{"openAPIV3Schema":{"description":"OverlappingRangeIPReservation is the Schema for the OverlappingRangeIPReservations API","properties":{"apiVersion":{"description":"APIVersion defines the versioned schema of this representation of an object.\nServers should convert recognized schemas to the latest internal value, and\nmay reject unrecognized values.\nMore info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources","type":"string"},"kind":{"description":"Kind is a string value representing the REST resource this object represents.\nServers may infer this from the endpoint the client submits requests to.\nCannot be updated.\nIn CamelCase.\nMore info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds","type":"string"},"metadata":{"type":"object"},"spec":{"description":"OverlappingRangeIPReservationSpec defines the desired state of OverlappingRangeIPReservation","properties":{"containerid":{"type":"string"},"ifname":{"type":"string"},"podref":{"type":"string"}},"required":["podref"],"type":"object"}},"required":["spec"],"type":"object"}},"served":true,"storage":true}]}}
    meta.helm.sh/release-name: harvester-crd
    meta.helm.sh/release-namespace: harvester-system
    objectset.rio.cattle.io/id: default-mcc-harvester-crd-cattle-fleet-local-system
  creationTimestamp: "2025-05-06T14:05:29Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
    objectset.rio.cattle.io/hash: d4a83267ddde6a8769c04362d4a0e5605db9baa7
  name: overlappingrangeipreservations.whereabouts.cni.cncf.io
  resourceVersion: "623709"
  uid: 19decf91-ad09-432f-abf3-22f227e01e66
```

#### Additional documentation or context

Vicente-Cheng reviewed (dismissed):
lgtm, thanks for the enhancement!

starbops reviewed (dismissed):
LGTM, thank you.

Yu-Jack committed:
feat: move dependency crds to harvester-crd folder

With takeOwnerShip field https://github.com/harvester/harvester-installer/blob/90668931a78c860f309487fe100be5ef868c3f81/pkg/config/templates/rancherd-10-harvester.yaml#L235
We can use that to manage dependency CRD.

Signed-off-by: Jack Yu <jack.yu@suse.com>

w13915984028 reviewed (commented):
Another open question:

After this change, there is no version info directly reading from on `deploy/charts/harvester-crd/templates` 

The current `harvester` chart has.
![image](https://github.com/user-attachments/assets/5b5079fe-3194-4a66-aace-81882ce49152)

We need a solution, either add the version info into the file name; or use another chart file to record the info. 

The version info helps the furture CRD bumping, validating and others.

WDYT @Yu-Jack , thanks.

Yu-Jack committed:
chore: add version as suffix

Signed-off-by: Jack Yu <jack.yu@suse.com>

w13915984028 reviewed (approved):
LGTM, thanks for the kind fix.

Harvester README.md needs an update, e.g.
https://github.com/harvester/harvester/blob/master/deploy/charts/harvester/README.md#chart-details

And harvester-crd needs a new README.md to describe the bump steps ...

ibrokethecloud reviewed (approved):
lgtm. thanks.
